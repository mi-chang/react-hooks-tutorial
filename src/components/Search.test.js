import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Search from './Search';

describe('Search', () => {
    it('handles SearchInputChanges', () => {
        const { component, getByLabelText } = render(<Search />);
        
        expect(component).toMatchSnapshot();

        const textbox = getByLabelText('Title');
        expect(textbox.value).toBe('');
        fireEvent.change(textbox, { target: { value: 'abc' } });
        expect(textbox.value).toBe('abc');

        expect(component).toMatchSnapshot();
    });

    it('calls search', () => {
        const search = jest.fn();
        const { component, getByLabelText, getByText } = render(<Search search={search} />);
        
        expect(component).toMatchSnapshot();

        const textbox = getByLabelText('Title');
        expect(textbox.value).toBe('');
        fireEvent.change(textbox, { target: { value: 'abc' } });
        expect(textbox.value).toBe('abc');
        expect(component).toMatchSnapshot();

        expect(search).toHaveBeenCalledTimes(0);
        fireEvent.click(getByText('SEARCH'));
        expect(search).toBeCalledTimes(1);
        expect(search).toBeCalledWith('abc');
        expect(textbox.value).toBe('');
        expect(component).toMatchSnapshot();
    })
});
