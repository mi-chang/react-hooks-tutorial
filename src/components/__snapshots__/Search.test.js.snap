// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search calls search 1`] = `undefined`;

exports[`Search calls search 2`] = `undefined`;

exports[`Search calls search 3`] = `undefined`;

exports[`Search handles SearchInputChanges 1`] = `undefined`;

exports[`Search handles SearchInputChanges 2`] = `undefined`;
